syncano/models/custom_sockets_utils.py

```diff
@@ -0,0 +1,222 @@
+
+
+class DependencyTypeE(object):
+    """
+    The type of the dependency object used in the custom socket;
+    """
+    SCRIPT = 'script'
+
+
+class BaseCall(object):
+    """
+    Base class for call object.
+    """
+
+    call_type = None
+
+    def __init__(self, name, methods):
+        self.name = name
+        self.methods = methods
+
+    def to_dict(self):
+        if self.call_type is None:
+            raise SyncanoValueError('call_type not set.')
+        return {
+            'type': self.call_type,
+            'name': self.name,
+            'methods': self.methods
+        }
+
+
+class ScriptCall(BaseCall):
+    """
+    Script call object.
+
+    The JSON format is as follows (to_dict in the base class)::
+
+        {
+            'type': 'script',
+            'name': '<script_label>,
+            'methods': [<method_list>],
+        }
+
+    methods can be as follows:
+        * ['GET']
+        * ['*'] - which will do a call on every request method;
+    """
+    call_type = CallTypeE.SCRIPT
+
+
+class Endpoint(object):
+    """
+    The object which stores metadata about endpoints in custom socket;
+
+    The JSON format is as follows::
+
+        {
+            '<endpoint_name>: {
+                'calls': [
+                    <list of JSON format of Calls objects>
+                ]
+            }
+        }
+
+    """
+    def __init__(self, name):
+        self.name = name
+        self.calls = []
+
+    def add_call(self, call):
+        self.calls.append(call)
+
+    def to_endpoint_data(self):
+        return {
+            self.name: {
+                'calls': [call.to_dict() for call in self.calls]
+            }
+        }
+
+
+class BaseDependency(object):
+    """
+    Base dependency object;
+
+    On the base of the fields attribute - the JSON format of the dependency is returned.
+    The fields are taken from the dependency object - which can be Script (supported now).
+    """
+
+    fields = []
+    dependency_type = None
+    field_mapping = {}
+
+    def __init__(self, dependency_object):
+        self.dependency_object = dependency_object
+
+    def to_dependency_data(self):
+        if self.dependency_type is None:
+            raise SyncanoValueError('dependency_type not set.')
+        dependency_data = {'type': self.dependency_type}
+        dependency_data.update({field_name: getattr(
+            self.dependency_object,
+            self.field_mapping.get(field_name, field_name)
+        ) for field_name in self.fields})
+        return dependency_data
+
+
+class ScriptDependency(BaseDependency):
+    """
+    Script dependency object;
+
+    The JSON format is as follows::
+        {
+            'type': 'script',
+            'runtime_name': '<runtime name defined in RuntimeChoices>',
+        }
+    """
+    dependency_type = DependencyTypeE.SCRIPT
+    fields = [
+        'runtime_name',
+        'name',
+        'source'
+    ]
+
+    field_mapping = {'name': 'label'}
```

Comment:
That's not exactly correct.
ScriptDependency is actually a dependency of ScriptEndpoint. 
It goes like this:

defined by name, runtime_name and source. `name` is used to create/get ScriptEndpoint. runtime_name and source are used for a underlying Script.
So I guess, ideally one could define it like:
```
ScriptDependency(
    name='custom_script',
    Script(
        runtime_name=RuntimeChoices.PYTHON_V5_0,
        source='print("custom_script")'
    )
)

ScriptDependency(
    name='custom_script',
    Script.please.first()
)

ScriptDependency(
    ScriptEndpoints.please.first()
)
```